agent/stats/service_connect_linux_test.go

```diff
@@ -43,26 +45,94 @@ func TestRetrieveServiceConnectMetrics(t *testing.T) {
 		LocalIPAddressUnsafe: "127.0.0.1",
 	}
 
-	// Set up a mock http sever on the statsUrlpath
-	statsMockurl := "127.0.0.1:9901"
-	r := mux.NewRouter()
-	r.HandleFunc("/stats/prometheus", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintln(w, "Service Connect Stats")
-	}))
+	var tests = []struct {
+		stats         string
+		expectedStats []*ecstcs.GeneralMetricsWrapper
+	}{
+		{
+			stats: "# TYPE MetricFamily1 counter\nMetricFamily1{dimensionA=\"value1\", dimensionB=\"value2\"} 1\n" +
+				"# TYPE MetricFamily2 counter\nMetricFamily2{dimensionB=\"value2\", dimensionA=\"value1\"} 1\n",
+			expectedStats: []*ecstcs.GeneralMetricsWrapper{
+				{
+					Dimensions: []*ecstcs.Dimension{
+						{
+							Key:   aws.String("dimensionA"),
+							Value: aws.String("value1"),
+						}, {
+							Key:   aws.String("dimensionB"),
+							Value: aws.String("value2"),
+						}},
+					GeneralMetrics: []*ecstcs.GeneralMetric{
+						{
+							MetricCounts: []*int64{aws.Int64(1)},
+							MetricName:   aws.String("MetricFamily1"),
+							MetricValues: []*float64{aws.Float64(1)},
+						},
+						{
+							MetricCounts: []*int64{aws.Int64(1)},
+							MetricName:   aws.String("MetricFamily2"),
+							MetricValues: []*float64{aws.Float64(1)},
+						},
+					},
+				},
+			},
+		},
+		{
+			stats: "# TYPE MetricFamily2 histogram\nMetricFamily2{dimensionX=\"value1\", dimensionY=\"value2\", le=\"0.5\"} 1\nMetricFamily2{dimensionX=\"value1\", dimensionY=\"value2\", le=\"1\"} 2\nMetricFamily2{dimensionX=\"value1\", dimensionY=\"value2\", le=\"5\"} 3\n",
```

Comment:
what happened with the `1`, `2`, and `3` values in the input? Should we assert something there?

e.g. `MetricFamily2{dimensionX=\"value1\", dimensionY=\"value2\", le=\"5\"} 3` <- this

Reply:
They got converted into Cloudwatch disjoint counts.  If you look at the expected stats
```
MetricCounts: []*int64{aws.Int64(1), aws.Int64(1), aws.Int64(1)}
```

Reply:
duh! my bad.